### PR TITLE
Marketplace version fix

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | https://github.com/mautic/marketplace-allowlist/pull/3
| Issue(s) addressed                     | 

#### Description:

With the release of the Custom Objects plugin we found that the Marketplace could handle some things better. So this PR fixes:
1. It takes the latest stable version and fallbacks to a feature branch if stable release does not exists.
2. Lets developers set a human readable name. Composer.json allows only for vendor/repository as the name.
3. The Issue Tracker button was not working. When in there I also added the wiki/documentation button.
4. I reordered the buttons a little. The install and the Close button is visible all the time, the Issue Tracker and Documentation are hidden in the context menu. It used to show the Install and the Issue Tracker button and Close being in the context menu.
5. Adds new "Clear Cache" button to the list of packages as I got an issue during development that an empty array got cached and I had no option to clear the cache other than waiting for an hour so the cache would invalidate. I think it could be useful for users too if they wait for new version of a package for example. And we should cache more stuff in the future.
6. Related to the point 5., I noticed that if there are no packages available then the message does not make sense. I fixed that like so:
Screen Shot 2022-05-31 at 10 39 06

Detail before:
Screen Shot 2022-06-01 at 13 27 19
Detail after:
Screen Shot 2022-06-01 at 13 28 43
Notice: the buttons are in different order, the issue tracker button works, the versions do not have a link unless they are stable as otherwise the link goes to a 404 page. The plugin name is more human readable. The version release date contains time.

List before:
Screen Shot 2022-06-01 at 13 27 35
List after:
Screen Shot 2022-06-01 at 13 28 30
Notice: The Custom Objects plugin name is human readable, the Clear Cache button appeared.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
8. Browse the Marketplace and notice the fixes in the description.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11197"></a>

### CODE REVIEW

File: app/bundles/MarketplaceBundle/Translations/en_US/messages.ini [hunk 1]
```diff
@@ -46,4 +47,8 @@ marketplace.package.remove.failed="Removing the package has failed. Please check
 marketplace.package.remove.html.failed="Something went wrong while removing <strong>%packagename%</strong>. This is the error:"
 marketplace.package.remove.html.in.progress="<strong>%packagename%</strong> is being removed. This might take a while..."
 marketplace.package.remove.html.success="Successfully removed <strong>%packagename%</strong>!"
-marketplace.package.remove.html.success.continue="Back to marketplace overview"
\ No newline at end of file
+marketplace.package.remove.html.success.continue="Back to marketplace overview"
+marketplace.latest.version.missing="Could not find any version of this package. Please try again later."
+marketplace.noresults.tip="There are no packages available for your version of Mautic. Try upgrading to a newer version."
+marketplace.clear.cache="Clear Cache"
+marketplace.clear.cache.tooltip="Some data are temporarily stored to speed up loading the Marketplace. Clear the cache to refresh the stored data."
```
Comment (RCheesley) [inline app/bundles/MarketplaceBundle/Translations/en_US/messages.ini]:
```suggestion
marketplace.clear.cache.tooltip="Some data is temporarily stored to speed up loading the Marketplace. Clear the cache to refresh the stored data."
```

File: app/bundles/MarketplaceBundle/Translations/en_US/messages.ini [hunk 2]
```diff
@@ -46,4 +47,8 @@ marketplace.package.remove.failed="Removing the package has failed. Please check
 marketplace.package.remove.html.failed="Something went wrong while removing <strong>%packagename%</strong>. This is the error:"
 marketplace.package.remove.html.in.progress="<strong>%packagename%</strong> is being removed. This might take a while..."
 marketplace.package.remove.html.success="Successfully removed <strong>%packagename%</strong>!"
-marketplace.package.remove.html.success.continue="Back to marketplace overview"
\ No newline at end of file
+marketplace.package.remove.html.success.continue="Back to marketplace overview"
```
Comment (RCheesley) [inline app/bundles/MarketplaceBundle/Translations/en_US/messages.ini]:
```suggestion
marketplace.package.remove.html.success.continue="Back to Marketplace overview"
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Thanks for the PR @escopecz - a few grammar changes and I spotted this issue with the batch select - something weird, or maybe no options? Either way I think we need to decide what behaviours are meant to be happening here:

https://watch.screencastify.com/v/j6UHIy6bDMLNhJB8KrSa

As a user I was expecting to find there the ability to select several plugins and have them installed, in case that is helpful :)

**Review by kuzmany — APPROVED** [review]
I did test in my ddev environment and looks like works pretty well. We can merge it, any improvement to marketplace welcome 👍

**Pull request merged**